### DISCUSSION TITLE
How to catch custom Exception inside Async consumer

### DISCUSSION THREAD

**Discussion by khaihoan1** [django/channels#1874] [Q&A]
Hi, I have an async consumer that has the method `chatroom()` that tries to call another method `_can_access_room()`.  In `_can_access_room()` there's a case it raises a custom Exception (inherit from Exception). But I cannot catch the custom Exception in `chatroom()`. The log shows that it raised, and WebSocket disconnects. I set a breakpoint on the line that catches the exception but it doesn't stop there. But if I catch the base Exception, it works. Please help, thanks ^^

Here's my code:

```
async def chatroom(self, data: dict):
    # check room in group or not?
    if group_from_room_id(data.get('room_id')) in self.groups:
        await self.chatroom_send(data)
    else:
        try:
            a = await self._can_access_room(data)
        except CustomException as e:
            print('ahihi')
@database_sync_to_async
    def _can_access_room(self, room_data: dict):
        room_id = room_data.get('room_id')
        chat_room: ChatRoom = ChatRoom.objects.filter(id=room_id).first()
        if not chat_room:
            raise custom_exceptions.RoomDoesNotExist
```
Here's the log:

```
...
, line 455, in thread_handler
    return func(*args, **kwargs)
  File "***********/consumers/chat_room_consumer.py", line 153, in _can_access_room
    raise custom_exceptions.RoomDoesNotExist
common.custom_exceptions.RoomDoesNotExist
WebSocket DISCONNECT /ws/chat/winners [127.0.0.1:49452]
```

**Comment by carltongibson** [comment]
I'd try adding a `except BaseException` to see what comes out of the `_can_access_room` call. 🤔

**Reply by khaihoan1** [reply]
It turns out it's my typo. I raise CustomException from a py module that is duplicated from another module that I forgot to delete, and I catch the CustomException from the old one. Sorry for wasting your time.
